clean_helpers/deduplication.py

```diff
@@ -80,3 +47,76 @@ def dedup_template(ds: Dataset, num_proc: int, batch_size: int) -> Dataset:
         )
 
     return dedup_template
+
+
+def dedup_document(ds: Dataset, num_proc: int, batch_size: int) -> Dataset:
+    hashed_documents = ds.map(
+        lambda batch: {**batch, "hash": get_hash(batch["text"])},
+        num_proc=num_proc,
+        batched=True,
+        batch_size=batch_size,
+    )
+
+    hashes = set()
+
+    return hashed_documents.filter(
+        lambda hashes_: [is_new_hash(hash_, hashes) for hash_ in hashes_],
+        num_proc=1,  # VERY IMPORTANT: hashes will be updated, and is not thread safe.
+        input_columns=["hash"],
+        batched=True,
+        batch_size=batch_size,
+    ).remove_columns("hash")
+
+
+# =========== HELPERS ===============
+
+def get_hash(texts: List[str]) -> List[str]:
+    """Get hash of content field."""
+    return [hashlib.md5(text.encode("utf-8")).hexdigest() for text in texts]
```

Comment:
what do you think about adding a `strip` before `encode` just to make sure we ignore surrounding whitespace.